### DISCUSSION TITLE
Garage Door Opener with Auto-Close Function

### DISCUSSION THREAD

**Discussion by SMMMN** [mongoose-os-apps/shelly-homekit#1475] [Ideas]
Hi all, thanks for the great software. 

I live in a multi-family house and the garage can be opened with a remote control. The closing process happens automatically after 2 minutes.
I have two cables connected to the remote and I can open the garage in switch mode with the auto-off function. It would be practical if the Switch is now displayed as a garage so that I can also say "Open garage" to Siri. 

Is this already possible?

Many greetings
Simon

**Comment by timoschilling** [comment]
We have support for garage door opener already. But no auto close functioning at the moment.

**Comment by Pixel-Chris** [comment]
the most garade door opener have an integrated autoclose function
or you realise it in homekit with an automation

**Comment by raddi** [comment]
+1 
A lot of car gates do have an auto close function, it would be nice to implement this feature to set a timer for closing without pressing a button :) 
Thanks!

**Comment by andyblac** [comment]
can Auto Off be tweaked to be used for this ?

**Comment by timoschilling** [comment]
Garage door has no auto off, but porting the auto off logic should be work

**Comment by rojer** [comment]
yep, makes sense to add auto-close timeout.

**Comment by raddi** [comment]
Anything new?

**Comment by rojer** [comment]
nope
